### ISSUE TITLE
:: is sometimes a variable name

### ISSUE THREAD

**Issue by nbaum** [JuliaLang/julia#18561]
`::` behaves like an identifier except it can't be assigned to.

`f(::) = ::` works and defines the identity function.

`f(::Any) = ::` works, but raises `UndefVarError` when called.

`:: = 42` is invalid syntax. (Unexpected "=".)

I think that `::` _should be_ semantically equivalent to `::Any` although possibly not parsed the same: macros might find the difference useful.

**Comment by simonster** [comment]
```julia
(::)(::) = (::)(::)
(::)((::)::typeof(::)) = ::
((::)(::)(::)::typeof(::))(Type){typeof((::)(::))}
```

**Comment by nbaum** [comment]
> Your scientists were so preoccupied with whether or not they could, they didn’t stop to think if they should.

**Comment by fredrikekre** [comment]
same as #18912 and fixed by #22947
